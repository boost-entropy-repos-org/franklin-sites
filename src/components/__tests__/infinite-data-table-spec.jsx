import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import InfiniteDataTable from '../infinite-data-table';

Enzyme.configure({ adapter: new Adapter() });

describe('DataTable component', () => {
  const columns = [
    {
      label: 'Column 1',
      name: 'col1',
      render: row => <span>{row.fieldValue1.value}</span>,
      width: 40,
    },
    {
      label: 'Column 2',
      name: 'col2',
      render: row => <span>{row.fieldValue2.value}</span>,
      sorted: 'ascend',
      sortable: true,
      width: 40,
    },
  ];
  const data = [
    {
      id: 'blah1',
      fieldValue1: {
        value: 'Some data 1',
      },
      fieldValue2: {
        value: 'Some data 2',
      },
    },
    {
      id: 'blah2',
      fieldValue1: {
        value: 'Some data A',
      },
      fieldValue2: {
        value: 'Some data B',
      },
    },
  ];

  const props = {
    selectable: true,
    fixedColumnCount: 0,
    fixedRowCount: 1,
    showHeader: true,
    showRowNumbers: true,
    numberResultsPerRequest: 10,
    totalNumberRows: 30,
    idKey: 'accessionId',
    onLoadMoreRows: () => {},
    columns,
    data,
  };

  test('should render', () => {
    const component = renderer.create(<InfiniteDataTable {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
