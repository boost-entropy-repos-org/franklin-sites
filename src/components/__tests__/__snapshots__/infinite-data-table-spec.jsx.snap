// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable component should render 1`] = `
<div
  style={
    Object {
      "height": 0,
      "overflow": "visible",
      "width": 0,
    }
  }
>
  <div
    className="data-table-infinite"
  >
    <div
      style={
        Object {
          "height": 400,
          "overflow": "visible",
          "width": 400,
        }
      }
    >
      <div
        style={
          Object {
            "height": 0,
            "position": "relative",
            "width": 400,
          }
        }
      >
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid"
          onScroll={[Function]}
          role="grid"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": 0,
              "left": 0,
              "overflowX": "hidden",
              "overflowY": "hidden",
              "position": "absolute",
              "top": 0,
              "width": 400,
              "willChange": "transform",
            }
          }
          tabIndex={null}
        />
      </div>
      <div
        style={
          Object {
            "height": 400,
            "overflow": "visible",
            "position": "relative",
            "width": 400,
          }
        }
      >
        <div
          aria-label="grid"
          aria-readonly={true}
          className="ReactVirtualized__Grid"
          onScroll={[Function]}
          role="grid"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "boxSizing": "border-box",
              "direction": "ltr",
              "height": 400,
              "left": 0,
              "overflowX": "hidden",
              "overflowY": "hidden",
              "position": "absolute",
              "width": 400,
              "willChange": "transform",
            }
          }
          tabIndex={0}
        >
          <div
            className="ReactVirtualized__Grid__innerScrollContainer"
            role="rowgroup"
            style={
              Object {
                "height": 0,
                "maxHeight": 0,
                "maxWidth": 80,
                "overflow": "hidden",
                "pointerEvents": "",
                "position": "relative",
                "width": 80,
              }
            }
          >
            <div
              className="table-data table-row--odd"
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 40,
                }
              }
            >
              <span>
                Some data A
              </span>
            </div>
            <div
              className="table-data table-row--odd"
              style={
                Object {
                  "height": 0,
                  "left": 40,
                  "position": "absolute",
                  "top": 0,
                  "width": 40,
                }
              }
            >
              <span>
                Some data B
              </span>
            </div>
            <div
              className="table-data table-row--even"
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 40,
                }
              }
            >
              Loading...
            </div>
            <div
              className="table-data table-row--even"
              style={
                Object {
                  "height": 0,
                  "left": 40,
                  "position": "absolute",
                  "top": 0,
                  "width": 40,
                }
              }
            >
              Loading...
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
